src/e2e/java/teammates/e2e/cases/e2e/InstructorCourseDetailsPageE2ETest.java

```diff
@@ -0,0 +1,126 @@
+        if (!TestProperties.isDevServer()) {
+            student.email = TestProperties.TEST_STUDENT1_ACCOUNT;
+        }
+
+        removeAndRestoreDataBundle(testData);
+        course = testData.courses.get("ICDetailsE2eT.CS2104");
+        fileName = "/" + course.getId() + "_studentList.csv";
+    }
+
+    @BeforeClass
+    public void classSetup() {
+        deleteDownloadsFile(fileName);
+    }
+
+    @Test
+    public void allTests() {
+        AppUrl detailsPageUrl = createUrl(Const.WebPageURIs.INSTRUCTOR_COURSE_DETAILS_PAGE)
+                .withUserId(testData.instructors.get("ICDetailsE2eT.instr").googleId)
+                .withCourseId(course.getId());
+        InstructorCourseDetailsPage detailsPage = loginAdminToPage(detailsPageUrl, InstructorCourseDetailsPage.class);
+
+        ______TS("verify loaded details");
+        InstructorAttributes[] instructors = {
+                testData.instructors.get("ICDetailsE2eT.instr"),
+                testData.instructors.get("ICDetailsE2eT.instr2"),
+        };
+        StudentAttributes[] students = {
+                testData.students.get("alice.tmms@ICDetailsE2eT.CS2104"),
+                testData.students.get("charlie.tmms@ICDetailsE2eT.CS2104"),
+                testData.students.get("benny.tmms@ICDetailsE2eT.CS2104"),
+                testData.students.get("danny.tmms@ICDetailsE2eT.CS2104"),
+        };
+
+        verifyCourseDetails(detailsPage, course, instructors, students);
+        detailsPage.verifyNumStudents(4);
+        detailsPage.verifyStudentDetails(students);
+
+        ______TS("send invite");
+        detailsPage.sendInvite(student);
+
+        detailsPage.verifyStatusMessage("An email has been sent to " + student.getEmail());
+        String expectedEmailSubject = "TEAMMATES: Invitation to join course ["
+                + course.getName() + "][" + course.getId() + "]";
+        verifyEmailSent(student.getEmail(), expectedEmailSubject);
+
+        ______TS("remind all students to join");
+        detailsPage.remindAllToJoin();
+
+        detailsPage.verifyStatusMessage("Emails have been sent to unregistered students.");
+        verifyEmailSent(student.getEmail(), expectedEmailSubject);
+
+        ______TS("download student list");
+        detailsPage.downloadStudentList();
+        String status = student.googleId.isEmpty() ? "Yet to Join" : "Joined";
+        String lastName = student.getName().split(" ")[1];
+        String[] studentInfo = { student.getTeam(), student.getName(), lastName, status, student.getEmail() };
+        List<String> expectedContent = Arrays.asList("Course ID," + course.getId(),
+                "Course Name," + course.getName(), String.join(",", studentInfo));
+        verifyDownloadedFile(fileName, expectedContent);
+
+        ______TS("delete student");
+        detailsPage.sortByName();
+        detailsPage.sortByStatus();
+        StudentAttributes[] studentsAfterDelete = { students[3], students[0], students[2] };
```

Comment:
This one is really your call! I would actually sort and remove the student.

Reply:
I think since it is an array it is simpler to just make a new one. Might swap everything to use Lists instead at a later date.